### PR TITLE
`azurerm_app_service_connection`, `azurerm_spring_cloud_connection`: add support for `key_vault_id`

### PR DESCRIPTION
**Pull request by jiaweitao001**
* Test results:
```
--- PASS: TestAccServiceConnectorAppService_complete(827.32s)
--- PASS: TestAccServiceConnectorSpringCloud_complete (1132.22s)
PASS
```

### CODE REVIEW

File: website/docs/r/service_connector_app_service.html.markdown [hunk 1]
```diff
@@ -109,6 +109,8 @@ The following arguments are supported:
 
 * `vnet_solution` - (Optional) The type of the VNet solution. Possible values are `serviceEndpoint`, `privateLink`.
 
+* `key_vault_id` - (Optional) The key vault id to store secret.
```
Comment (katbyte) [inline website/docs/r/service_connector_app_service.html.markdown:112]: should this be
```suggestion
* `secret_store_key_vault_id` - (Optional) The key vault id to store secret.
```

Comment (katbyte) [reply website/docs/r/service_connector_app_service.html.markdown:112]: is this to store the secret store in? load the secret store from? could we expand the description here some

File: website/docs/r/service_connector_spring_cloud.html.markdown [hunk 2]
```diff
@@ -114,6 +114,8 @@ The following arguments are supported:
 
 * `vnet_solution` - (Optional) The type of the VNet solution. Possible values are `serviceEndpoint`, `privateLink`.
 
+* `secret_store` - (Optional) The key vault id to store secret.
```
Comment (katbyte) [inline website/docs/r/service_connector_spring_cloud.html.markdown:117]: and here

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/service_connector_app_service.html.markdown (hunk 1 above)
Comment (jiaweitao001) [reply website/docs/r/service_connector_app_service.html.markdown]: Will change.

File: website/docs/r/service_connector_spring_cloud.html.markdown (hunk 2 above)
Comment (jiaweitao001) [reply website/docs/r/service_connector_spring_cloud.html.markdown]: Will change.

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @jiaweitao001 - we have test failures to fix:

```
------- Stdout: -------
=== RUN   TestAccServiceConnectorSpringCloud_complete
=== PAUSE TestAccServiceConnectorSpringCloud_complete
=== CONT  TestAccServiceConnectorSpringCloud_complete
    testcase.go:110: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating Scoped Linker (Resource Uri: "/subscriptions/*******/resourceGroups/acctestRG-221024172621009059/providers/Microsoft.AppPlatform/Spring/testspringcloudservice-m4wgc/apps/testspringcloud-m4wgc/deployments/deploy-m4wgc"
        Linker Name: "acctestserviceconnector221024172621009059"): polling after LinkerCreateOrUpdate: Code="InvalidVNetSolution" Message="Execution failed. Network solution type is not available: Source is not deployed in VNet"
        
          with azurerm_spring_cloud_connection.test,
          on terraform_plugin_test.tf line 82, in resource "azurerm_spring_cloud_connection" "test":
          82: resource "azurerm_spring_cloud_connection" "test" {
        
        creating Scoped Linker (Resource Uri:
        "/subscriptions/*******/resourceGroups/acctestRG-221024172621009059/providers/Microsoft.AppPlatform/Spring/testspringcloudservice-m4wgc/apps/testspringcloud-m4wgc/deployments/deploy-m4wgc"
        Linker Name: "acctestserviceconnector221024172621009059"): polling after
        LinkerCreateOrUpdate: Code="InvalidVNetSolution" Message="Execution failed.
        Network solution type is not available: Source is not deployed in VNet"
--- FAIL: TestAccServiceConnectorSpringCloud_complete (1180.57s)
```
and
```
------- Stdout: -------
=== RUN   TestAccServiceConnectorAppServiceCosmosdb_update
=== PAUSE TestAccServiceConnectorAppServiceCosmosdb_update
=== CONT  TestAccServiceConnectorAppServiceCosmosdb_update
    testcase.go:110: Step 1/3 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_linux_web_app.test will be updated in-place
          ~ resource "azurerm_linux_web_app" "test" {
              ~ app_settings                      = {
                  - "AZURE_COSMOS_LISTCONNECTIONSTRINGURL" = "https://management.azure.com/subscriptions/*******/resourceGroups/acctestRG-221024172621003635/providers/Microsoft.DocumentDB/databaseAccounts/acctestaccn7o84/listConnectionStrings?api-version=2021-04-15" -> null
                  - "AZURE_COSMOS_RESOURCEENDPOINT"        = "[https://acctestaccn7o84.documents.azure.com:443/](https://acctestaccn7o84.documents.azure.com/)" -> null
                  - "AZURE_COSMOS_SCOPE"                   = "https://management.azure.com/.default" -> null
                }
                id                                = "/subscriptions/*******/resourceGroups/acctestRG-221024172621003635/providers/Microsoft.Web/sites/linuxwebappn7o84"
                name                              = "linuxwebappn7o84"
                tags                              = {}
                # (17 unchanged attributes hidden)
        
        
              - identity {
                  - identity_ids = [] -> null
                  - principal_id = "9bb0d676-b804-4acf-92d1-83925627520f" -> null
                  - tenant_id    = "*******" -> null
                  - type         = "SystemAssigned" -> null
                }
        
                # (2 unchanged blocks hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccServiceConnectorAppServiceCosmosdb_update (1080.04s)
FAIL
```

File: internal/services/serviceconnector/service_connector_app_service_resource.go [hunk 3]
```diff
@@ -264,6 +283,13 @@ func (r AppServiceConnectorResource) Update() sdk.ResourceFunc {
 				linkerProps.VNetSolution = &vnetSolution
 			}
 
+			if d.HasChange("secret_store") {
```
Comment (jackofallops) [inline internal/services/serviceconnector/service_connector_app_service_resource.go:286]: Should be?
```suggestion
			if d.HasChange("secret_store_key_vault_id") {
```

File: internal/services/serviceconnector/service_connector_spring_cloud_resource.go [hunk 4]
```diff
@@ -267,6 +286,13 @@ func (r SpringCloudConnectorResource) Update() sdk.ResourceFunc {
 				linkerProps.VNetSolution = &vnetSolution
 			}
 
+			if d.HasChange("secret_store") {
```
Comment (jackofallops) [inline internal/services/serviceconnector/service_connector_spring_cloud_resource.go:289]: as above:
```suggestion
			if d.HasChange("secret_store_key_vault_id") {
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @jiaweitao001 
I spotted a couple minor changes required, could you take a look?

Also, did you get chance to look into the test failures Katbyte highlighted?

Thanks!

**Comment by jiaweitao001** [comment]
The related acc tests are failing of upstream services, will close this PR for now. I'll reopen it once the issue is fixed.

**Pull request closed without merging**